### PR TITLE
Add quality declaration console_bridge_vendor

### PR DESCRIPTION
**Pull request by Blast545**
This is a quality declaration for console_bridge_vendor as described in the proposed REP-2004 (see ros-infrastructure/rep#218).

EDIT: This PR contains the current quality declaration claim for this package, in this case, QL4.

Checklist in #7 can be checked first to sort out missing parts of the declaration.

Signed-off-by: Jorge Perez <jjperez@ekumenlabs.com>

### CODE REVIEW

**Comment by ahcorde** [comment]
This PR needs some rework with the current status of the QD

File: Quality_Declaration.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+`console_bridge_vendor` uses `semver` according to the recommendation for ROS Core packages in the [ROS 2 Developer Guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#versioning).
+
+### Version Stability [1.ii]
+
+`console_bridge_vendor` is at or above a stable version, i.e. `>= 1.0.0`.
+
+### Public API Declaration [1.iii]
+
+External dependency is loaded downloaded from upstream and there is no public API declaration hosted for `libconsole-bridge-dev`.
```
Comment (brawner) [inline Quality_Declaration.md]:
```suggestion
This is a vendor package for `libconsole-bridge-dev` and as such does not declare its own API.
```

File: Quality_Declaration.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+### API Stability Policy [1.iv]
+
+`console_bridge_vendor` is pinned to a `libconsole-bridge-dev` version (Currently 0.4.1), so it will not break public API within a released ROS distribution, i.e. no major releases once the ROS distribution is released.
```
Comment (brawner) [inline Quality_Declaration.md]: This sentence is not clear. Try something like:

"`console_bridge_vendor` does not have an API of its own, and will limit its dependency on `libconsole-bridge-dev` to API stable releases within a released ROS distribution."

File: Quality_Declaration.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+### ABI Stability Policy [1.v]
+
+`console_bridge_vendor` is pinned to a `libconsole-bridge-dev` version (Currently 0.4.1), so it will not break public ABI within a released ROS distribution, i.e. no major releases once the ROS distribution is released.
```
Comment (brawner) [inline Quality_Declaration.md]: Same.

File: Quality_Declaration.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+### License [3.iii]
+
+The license for `console_bridge_vendor` is Apache 2.0, and a summary is in each source file, the type is declared in the [`package.xml`](./package.xml) manifest file, and a full copy of the license is in the `LICENSE` file.
+
+### Copyright Statements [3.iv]
+
+The copyright holders each provide a statement of copyright in each source code file in `console_bridge_vendor`.
+
+## Testing [4]
```
Comment (brawner) [inline Quality_Declaration.md]: Since this is such a basic package, I would say that that feature testing, api testing, code coverage and performance testing are not needed. Otherwise, it kind of looks like they are missing, when they are really just irrelevant.

File: Quality_Declaration.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+## Dependencies [5]
+
+### Direct and Optional Runtime ROS Dependencies [5.i]/[5.ii]
+
+`console_bridge_vendor` does not have direct/optional runtime ROS dependencies.
+
+### Direct Runtime non-ROS Dependency [5.iii]
+
+`console_bridge_vendor` depends directly on the external dependency `libconsole-bridge-dev`, this one is qualified as quality level 4 as described thoroughly [QD libconsole-bridge-dev](https://github.com/ros/console_bridge/blob/master/Quality_Declaration.md).
```
Comment (brawner) [inline Quality_Declaration.md]:
```suggestion
`console_bridge_vendor` depends directly on the external dependency `libconsole-bridge-dev`, which is qualified as quality level 4 in its [Quality Declaration](https://github.com/ros/console_bridge/blob/master/Quality_Declaration.md).
```

**Review by brawner — COMMENTED** [review]
Why is this not tested in CI? Which packages depend on it?

File: Quality_Declaration.md (hunk 4 above)
Comment (Blast545) [reply Quality_Declaration.md]: I agree to that for vendored packages, as the dependencies should address this (or argue why is not needed).

**Comment by Blast545** [comment]
DCO is not yet enabled for this package, should be enabled before merging

**Comment by Blast545** [comment]
> Why is this not tested in CI?

Probably is being tested but doesn't appears in the report because it doesn't have tests. I'll open another PR enabling linter/copyright tests to check it out.

File: Quality_Declaration.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+
+This is a vendor package for `libconsole-bridge-dev` and as such does not declare its own API.
+
+### API Stability Policy [1.iv]
+
+`console_bridge_vendor` does not have an API of its own, and will limit its dependency on `libconsole-bridge-dev` to API stable releases within a released ROS distribution.
+
+### ABI Stability Policy [1.v]
+
+`console_bridge_vendor` does not have an ABI of its own, and will limit its dependency on `libconsole-bridge-dev` to API stable releases within a released ROS distribution.
```
Comment (brawner) [inline Quality_Declaration.md]:
```suggestion
`console_bridge_vendor` does not have an ABI of its own, and will limit its dependency on `libconsole-bridge-dev` to ABI stable releases within a released ROS distribution.
```

File: Quality_Declaration.md [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+
+## Documentation [3]
+
+### Feature Documentation [3.i]
+
+There is some documentation on its latest [wiki](http://wiki.ros.org/console_bridge). This information might require to be updated and expanded further for the use of `console_bridge_vendor`.
+
+### Public API Documentation [3.ii]
+
+`console_bridge_vendor` does not have Public API Documentation.
```
Comment (brawner) [inline Quality_Declaration.md]: "`console_bridge_vendor` does not have an API and therefore does not require API documentation."

File: Quality_Declaration.md [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+
+### Copyright Statements [3.iv]
+
+The copyright holders each provide a statement of copyright in each source code file in `console_bridge_vendor`.
+
+## Testing [4]
+
+### Feature Testing [4.i]
+
+`console_bridge_vendor` does not have features testing. For vendored packages, this will be considered to be addressed in the external dependency.
```
Comment (brawner) [inline Quality_Declaration.md]:
```suggestion
`console_bridge_vendor` does not have feature testing. For vendored packages, this will be considered to be addressed in the external dependency.
```

File: Quality_Declaration.md [hunk 9]
```diff
@@ -0,0 +1,117 @@
+This document is a declaration of software quality for the `console_bridge_vendor` package, based on the guidelines in [REP-2004](https://www.ros.org/reps/rep-2004.html).
```
Comment (brawner) [inline Quality_Declaration.md]: I just noticed how you were naming all of these, I thought we were naming things QUALITY_DECLARATION.md?

File: Quality_Declaration.md [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+
+`console_bridge_vendor` does not have an API of its own, and will limit its dependency on `libconsole-bridge-dev` to API stable releases within a released ROS distribution.
+
+### ABI Stability Policy [1.v]
+
+`console_bridge_vendor` does not have an ABI of its own, and will limit its dependency on `libconsole-bridge-dev` to ABI stable releases within a released ROS distribution.
+
+### ABI and ABI Stability Within a Released ROS Distribution [1.vi]
+
+`console_bridge_vendor` will not break API nor ABI within a released ROS distribution, i.e. no major releases once the ROS distribution is released.
```
Comment (ahcorde) [inline Quality_Declaration.md]: I would rephrase this, if there is no API or ABI how are we going to break it?. I would say something like: "As there are not API or ABI issue in this package, this section doesn't applied"

**Review by ahcorde — CHANGES_REQUESTED** [review]
Rephase section: `ABI and ABI Stability Within a Released ROS Distribution`

File: QUALITY_DECLARATION.md [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+
+`console_bridge_vendor` does not have an API of its own, and will limit its dependency on `libconsole-bridge-dev` to API stable releases within a released ROS distribution.
+
+### ABI Stability Policy [1.v]
+
+`console_bridge_vendor` does not have an ABI of its own, and will limit its dependency on `libconsole-bridge-dev` to ABI stable releases within a released ROS distribution.
+
+### ABI and ABI Stability Within a Released ROS Distribution [1.vi]
+
+As `console_bridge_vendor` does not come with its own API/ABI, changes will not affect API/ABI stability.
```
Comment (brawner) [inline QUALITY_DECLARATION.md]:
```suggestion
As `console_bridge_vendor` does not have its own API/ABI, changes will not affect API/ABI stability.
```

Also during the above two sections, you already state that it will keep libconsole-bridge-dev pinned to stable releases within a ROS distribution. Either combine this section with those two, or move the text about the ROS distribution down here.

File: QUALITY_DECLARATION.md [hunk 12]
```diff
@@ -0,0 +1,117 @@
...
+
+This package uses DCO as its confirmation of contributor origin policy. More information can be found in [CONTRIBUTING](./CONTRIBUTING.md).
+
+### Peer Review Policy [2.iii]
+
+All pull requests will be peer-reviewed, check [ROS 2 Developer Guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#change-control-process) for additional information.
+
+### Continuous Integration [2.iv]
+
+This package is not being tested under ROS2 CI.
```
Comment (chapulina) [inline QUALITY_DECLARATION.md:49]: It's part of `ros2.repos` though: https://github.com/ros2/ros2/blob/master/ros2.repos#L170

File: Quality_Declaration.md (hunk 9 above)
Comment (Blast545) [reply Quality_Declaration.md]: Yes, we agreed on the standard after this was opened initially, I'll fix it for all my PRs.

File: QUALITY_DECLARATION.md (hunk 12 above)
Comment (Blast545) [reply QUALITY_DECLARATION.md:49]: The linters are not enabled on the `package.xml` file, I'll open another PR to fix this, already assigned myself in the spreadsheet.

File: Quality_Declaration.md (hunk 9 above)
Comment (Blast545) [reply Quality_Declaration.md]: Just checked, the latest commit of this PR already has this addressed.

File: QUALITY_DECLARATION.md [hunk 13]
```diff
@@ -0,0 +1,113 @@
...
+
+`console_bridge_vendor` does not have an ABI of its own, and will limit its dependency on `libconsole-bridge-dev` to ABI stable releases within a released ROS distribution.
+
+## Change Control Process [2]
+
+`console_bridge_vendor` follows the recommended guidelines for ROS Core packages in the [ROS 2 Developer Guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#change-control-process).
+
+### Change Requests [2.i]
+
+All changes will occur through a pull request, check [ROS 2 Developer Guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#change-control-process) for additional information.
```
Comment (brawner) [inline QUALITY_DECLARATION.md]:
```suggestion
All changes will occur through a pull request, check the [ROS 2 Developer Guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#change-control-process) for additional information.
```

**Review by brawner — APPROVED** [review]
Small nit.
